File: 이코테/c++/1이 될 때까지.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
	int N, K, cnt, target;
	cin >> N >> K;	
	
	while(true){		
		target = (N / K) * K;
		cnt += (N - target);
		N = target;
		if(N < K){
			break;
		}
		N /= K;
		cnt += 1;
	} 
	cnt += (N - 1);
	cout << cnt << endl;
}

File: 이코테 복습(python, c++, java)/그리디/java/_1이_될_때까지.java
package javastudy.이코테.그리디;

import java.util.Scanner;

public class _1이_될_때까지 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int cnt = 0;

        while (true) {
            int target = (n / k) * k;
            cnt += n - target;
            n = target;

            if (n < k) {
                cnt += n - 1;
                break;
            }

            n /= k;
            cnt += 1;
        }
        System.out.println(cnt);
    }
}
